### PR TITLE
MULE-18493: HTTP Connector leaking sockets in state CLOSE_WAIT when client uses HTTP/1.0

### PR DESCRIPTION
**Pull request by pabloperalta**

### CODE REVIEW

**Comment by pabloperalta** [comment]
--validate

**Comment by IvanAndresFritzler** [comment]
--validate

**Comment by IvanAndresFritzler** [comment]
--validate

File: modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/ResponseDeferringCompletionHandler.java [hunk 1]
```diff
@@ -98,9 +99,12 @@ public void completed(WriteResult result)
 
   private void doComplete()
   {
-    responseStatusCallback.responseSendSuccessfully();
-    ctx.notifyDownstream(RESPONSE_COMPLETE_EVENT);
-    resume();
+    if(!isCompleted) {
```
Comment (eze210) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/ResponseDeferringCompletionHandler.java:102]: Use an AtomicBoolean and compareAndSet

Comment (pabloperalta) [reply modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/ResponseDeferringCompletionHandler.java]: Hecho

**Pull request merged**